drop-storage/src/lib.rs

```diff
@@ -1,630 +1,532 @@
 pub mod error;
 pub mod types;
 
-use std::str::FromStr;
-
+use std::vec;
+
+use include_dir::{include_dir, Dir};
+use lazy_static::lazy_static;
+use r2d2::Pool;
+use r2d2_sqlite::SqliteConnectionManager;
+use rusqlite::{params, Transaction};
+use rusqlite_migration::Migrations;
 use slog::Logger;
-use sqlx::{sqlite::SqliteConnectOptions, SqliteConnection, SqlitePool};
 use types::{
     DbTransferType, IncomingPath, IncomingPathStateEvent, IncomingPathStateEventData, OutgoingPath,
     OutgoingPathStateEvent, OutgoingPathStateEventData, Transfer, TransferFiles,
     TransferIncomingPath, TransferOutgoingPath, TransferStateEvent,
 };
-use uuid::{fmt::Hyphenated, Uuid};
+use uuid::Uuid;
 
 use crate::error::Error;
 pub use crate::types::{Event, FileChecksum, TransferInfo, TransferType};
 
 type Result<T> = std::result::Result<T, Error>;
+type QueryResult<T> = std::result::Result<T, rusqlite::Error>;
 // SQLite storage wrapper
 pub struct Storage {
     _logger: Logger,
-    conn: SqlitePool,
+    pool: Pool<SqliteConnectionManager>,
+}
+
+static MIGRATIONS_DIR: Dir = include_dir!("$CARGO_MANIFEST_DIR/migrations");
+
+lazy_static! {
+    static ref MIGRATIONS: Migrations<'static> = Migrations::from_directory(&MIGRATIONS_DIR)
```

Comment:
I wouldn't say I like it. Why do we use static here? Can't we create the thing at the call site?

Reply:
good point, done